sendgrid/helpers/mail/decorators.py

```diff
@@ -0,0 +1,37 @@
+"""Contains decorators used in helper."""
+
+import inspect
+import six
+import wrapt
+
+
+def accepts(*types, **kwtypes):
+    """Decorator for verifying the type of method arguments."""
+    @wrapt.decorator
+    def wrapper(wrapped, instance, args, kwargs):
+        args_names = inspect.getargspec(wrapped)[0]
+        if instance is not None:
+            args_names = args_names[1:]
+
+        for (arg_val, expected_type, arg_name) in zip(args, types, args_names):
+            if expected_type is str:
+                expected_type = six.string_types
+
+            if not isinstance(arg_val, expected_type):
```

Comment:
I'm worried that this is going to cause more breaking changes than is functionally necessary. For example, with [this PR](https://github.com/sendgrid/sendgrid-python/pull/920) adding strict type checking broke a functional implementation.